error/s2n_errno.h

```diff
@@ -91,6 +96,8 @@ typedef enum {
     S2N_ERR_INVALID_MAX_FRAG_LEN,
     S2N_ERR_MAX_FRAG_LEN_MISMATCH,
     S2N_ERR_PROTOCOL_VERSION_UNSUPPORTED,
+    S2N_ERR_BAD_KEY_SHARE,
+    S2N_ERR_T_PROTO_END,
     /* S2N_ERR_T_INTERNAL */
```

Comment:
I wonder if there should be more padding and a better name for these categorization comments, it seems easy to overlook and one would just add an error to the bottom of the list

Reply:
Will add padding, but not going into better naming rabbit hole though :D